internal/trie/node/header.go

```diff
@@ -44,7 +45,8 @@ func (b *Branch) encodeHeader(writer io.Writer) (err error) {
 
 // encodeHeader creates the encoded header for the leaf.
 func (l *Leaf) encodeHeader(writer io.Writer) (err error) {
-	var header byte = 1 << 6
+	const nodeHeaderShift = 6
```

Comment:
should we maintain 2 `const` with the same name and value? why not put in the pkg level and reuse it?

Reply:
Totally, done! Good catch!